controllers/perconaserverformysql_controller.go

```diff
@@ -231,6 +246,18 @@ func (r *PerconaServerForMySQLReconciler) reconcileReplication(
 	ctx context.Context,
 	cr *apiv2.PerconaServerForMySQL,
 ) error {
+	l := log.FromContext(ctx).WithName("reconcileReplication")
+
+	sts := &appsv1.StatefulSet{}
+	if err := r.Get(ctx, client.ObjectKeyFromObject(orchestrator.StatefulSet(cr)), sts); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	if sts.Status.ReadyReplicas == 0 {
+		l.Info("orchestrator is not ready. skip", "ready", sts.Status.ReadyReplicas)
+		return nil
+	}
```

Comment:
We can check `cr.Status.Orchestrator.State`

Reply:
can you explain more @delgod?

Reply:
`cr.Status.Orchestrator.State` is a state at the end of the previous reconcile. could be 5 secs outdated

Reply:
This code assumes we can make queries to any of the orchestrator pods. It may be correct since we're running a single pod at the moment, but we'll be running orc as a cluster soon and we need to connect to the leader. I still think we need to check `cr.Status.Orchestrator.State` and accept the 5 secs delay.

Reply:
I think this code is naive. We can't just assume if Orchestrator statefulset has more than 0 ready pods, orchestrator is healthy. But this function is used to update the labels for primary pod and I see why @delgod doesn't want any delays. I'm OK to merge this as it's and improve afterwards.